ami/models/components/unet/decoder_block.py

```diff
@@ -0,0 +1,103 @@
+# Ref: https://github.com/facebookresearch/RCDM
+
+import torch
+import torch.nn as nn
+
+from .commons import AttentionBlock, ResBlock
+
+
+class UnetDecoderBlock(nn.Module):
+    """A block for Unet decoder."""
+
+    def __init__(
+        self,
+        in_channels: int,
+        in_channels_for_skip_connections: list[int],
```

Comment:
`in_channels_for_skip_connections`がlist[int]だとEncoderBlockと整合性取れなくない？
常にEncoderのresblockの出力チャネルは一定だから、blockとしての`in_channels_for_skip_connections`は常に一定にならない？
```suggestion
        in_channels_for_skip_connections: int,
```

Encoderのコード抜粋
```py
            self.resnet_layers.append(
                 ResBlock(
                     in_channels=(in_channels if res_block_i == 0 else out_channels),
                     emb_channels=timestep_embed_dim,
                     out_channels=out_channels,
                 )
             )
```

Reply:
EncoderのresblockとDecoderのresblockは必ずしも1対1対応してなくて、そのせいでskip connectionからやって来る特徴量のchannel数にズレがある場合があるから`in_channels_for_skip_connections`は`list[int]`じゃないといけない
(後で`Unet`のプルリク出すから詳しくはそこで見て欲しい)

Reply:
OK